File: DesignPatternsDotNet/PrototypeDesignPattern/AdventureBook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PrototypeDesignPattern
{
    class AdventureBook : IBookPrototype
    {
        public AdventureBook()
        {
            Console.WriteLine("An adventure book is bought");	
        }
        public object Clone() 
        {
            object aBook = null;
		    aBook = (AdventureBook)base.MemberwiseClone();		    
		    Console.WriteLine("An adventure book is cloned");
		    return aBook;
        }
    }
}

File: DesignPatternsJava/src/main/java/com/prototype/AdventureBook.java
package com.prototype;

public class AdventureBook implements IBookPrototype{
	public AdventureBook(){
		System.out.println("An adventure book is bought");
	}
	public IBookPrototype makeAClone() {
		AdventureBook aBook=null;
		try {
			aBook = (AdventureBook)super.clone();
		} catch (CloneNotSupportedException e) {
			e.printStackTrace();
		}
		System.out.println("An adventure book is cloned");
		return aBook;
	}
}
